### ISSUE TITLE
NaN gradient after loss.backward()

### ISSUE THREAD

**Issue by zhuole1025** [tianweiy/DMD2#56]
Hi Tianwei, thanks for your great work! While running the official training code to reproduce dmd2 on sdxl, I got all nan gradients after the first backward, although losses look very normal. Do you have any solution for that?

**Comment by sayakpaul** [comment]
I think this was resolved?

**Comment by 10RE** [comment]
I encountered the same problem with torch>=2.4.1 accelerate>=1.0.1. In my case nan gradients are caused by generating text embeddings from SDXLTextEncoder when there are multiple gpu.
Using `accelerator.prepare` to wrap around the text encoder in `main/sd_unified_model.py` solves the problem, no more nan gradients. But I'm getting the _post_backward_hook_state error mentioned in @ #41 , no idea how to fix this.

**Comment by Darius-H** [comment]
> I encountered the same problem with torch>=2.4.1 accelerate>=1.0.1. In my case nan gradients are caused by generating text embeddings from SDXLTextEncoder when there are multiple gpu. Using `accelerator.prepare` to wrap around the text encoder in `main/sd_unified_model.py` solves the problem, no more nan gradients. But I'm getting the _post_backward_hook_state error mentioned in @ [#41](https://github.com/tianweiy/DMD2/issues/41) , no idea how to fix this.

Have you solved this problem? I meet the same problem.
